### PR TITLE
feat(nix): add support for running external services through services-flake

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Described in #57 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Removes dependency on brew for installing and managing external services - redis and Postgres.

## How did you test it?
Locally.

- Make sure nix is installed
- `nix run .#ext-services`

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: docs/try_local_system.md [hunk 1]
```diff
@@ -166,6 +170,73 @@ Once the services have been confirmed to be up and running, you can proceed with
...
+- Add below line to your `~/.config/nix/nix.conf` or `/etc/nix/nix.conf`
+
+   ```
+   experimental-features = nix-command flakes
+   ```
+
+- References
+   - Install Nix - https://nixos.org/download/
+   - Enable flakes - https://nixos.wiki/wiki/Flakes#Other_Distros.2C_without_Home-Manager
+
```
Comment (srid) [inline docs/try_local_system.md:212]: I don't think we should recommend this. The DetSys installer is better.

```suggestion
```

File: docs/try_local_system.md [hunk 2]
```diff
@@ -166,6 +170,73 @@ Once the services have been confirmed to be up and running, you can proceed with
...
+*nixos-unified-template* provides an unified Nix configuration template for managing user applications and using `flakes`.
+
+- Head to [nixos-unified-template](https://github.com/juspay/nixos-unified-template?tab=readme-ov-file#on-non-nixos) and follow the instructions.
+
+#### 2. Using nix-installer [minimal]
+
+*nix-installer* provides a quick and reliable way of installing Nix with `flakes`.
+
+- Head to [nix-installer](https://github.com/DeterminateSystems/nix-installer) and follow the instructions.
```
Comment (srid) [inline docs/try_local_system.md:191]: We strive to document the process in external repos/pages, so the project README should be as simple/minimal as possible.

```suggestion
### Install nix

We recommend that you install Nix using [the DetSys nix-installer](https://nixos.asia/en/install), which automatically enables flakes.

As an **optional** next step, if you are interested in using Nix to manage your dotfiles and local packages, you can setup [nixos-unified-template](https://github.com/juspay/nixos-unified-template#on-non-nixos).
```

File: docs/try_local_system.md [hunk 3]
```diff
@@ -166,6 +170,73 @@ Once the services have been confirmed to be up and running, you can proceed with
...
+   - Enable flakes - https://nixos.wiki/wiki/Flakes#Other_Distros.2C_without_Home-Manager
+
+### Using external services through Nix
+
+Once Nix is installed, you can use it to manage external services via `flakes`. More services will be added soon.
+
+- Run below command in hyperswitch directory
+
+   ```shell
+   nix run .#ext-services
```
Comment (srid) [inline docs/try_local_system.md:220]: Works!

image

File: .gitignore [hunk 4]
```diff
@@ -263,4 +263,9 @@ result*
...
 
 # cypress credentials
-creds.json
\ No newline at end of file
+creds.json
+
+/.direnv
+
+# Nix services data
+data
```
Comment (srid) [inline .gitignore:271]:
```suggestion
/data
```

**Review by srid — APPROVED** [review]
I'd like to refactor this using https://nixos-unified.org/autowiring.html#flake-parts - but will do that in different PR.

This one can be merged.

File: flake.nix [hunk 5]
```diff
@@ -27,17 +31,48 @@
           devShells.default = pkgs.mkShell {
             name = "hyperswitch-shell";
             packages = with pkgs; [
+              fd
```
Comment (SanchithHegde) [inline flake.nix:34]: We don't necessarily _need_ `fd` as of now, we can add it later if required.

Comment (srid) [reply flake.nix:34]: FWIW, these are also included in `nixos-unified-template`

https://github.com/juspay/nixos-unified-template/blob/ed4bf673acdc802d06bfdd726494323ee2036f50/modules/home/packages.nix#L7-L12

Comment (kashif-m) [reply flake.nix]: Removed this for now, can be added later if needed!

**Review by SanchithHegde — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Pull request merged**